### ISSUE TITLE
🐛 BUG: Lit components throwing error in 0.20

### ISSUE THREAD

**Issue by jaydanurwin** [snowpackjs/astro#1313]
### What package manager are you using?

npm

### What operating system are you using?

Mac OS and Linux (tried both)

### Describe the Bug

I'll start by saying I'm not sure if this is an issue with Astro or the @astrojs/renderer-lit package.

Prior to 0.20 I wasn't having these issues with Lit web components being called through a script tag but now I'm getting the following errors/warnings below after upgrading. The Lit components in my app use the Material Web Components library which aren't 100% compatible with the way Astro handles Lit components by default. This forces me to use a `<script type="module">` tag for components using those UI components. 

The [framework-lit example](https://github.com/snowpackjs/astro/tree/main/examples/framework-lit) works fine with the Lit renderer and sample components until I go to use a script tag instead of the top level import and client:load parameter on the component. I wouldn't mind the warning so much but it's also keeping the Material Web Component from rendering. All that renders on page in their place is `[object Object]`

Where I'm confused is what changed between 0.19 and 0.20 for this break as I didn't see anything new that had to do with Lit or the Lit renderer.

Errors:
```
[17:20:50] [snowpack] lit-html/lib/shady-render.js: Unscannable package import found.
Snowpack scans source files for package imports at startup, and on every change.
But, sometimes an import gets added during the build process, invisible to our file scanner.
We'll prepare this package for you now, but should add "lit-html/lib/shady-render.js" to "knownEntrypoints"
in your config file so that this gets prepared with the rest of your imports during startup.
[17:20:50] [snowpack] lit-html/lit-html.js: Unscannable package import found.
Snowpack scans source files for package imports at startup, and on every change.
But, sometimes an import gets added during the build process, invisible to our file scanner.
We'll prepare this package for you now, but should add "lit-html/lit-html.js" to "knownEntrypoints"
in your config file so that this gets prepared with the rest of your imports during startup.
[17:20:50] [snowpack] lit-element: Unscannable package import found.
Snowpack scans source files for package imports at startup, and on every change.
But, sometimes an import gets added during the build process, invisible to our file scanner.
We'll prepare this package for you now, but should add "lit-element" to "knownEntrypoints"
in your config file so that this gets prepared with the rest of your imports during startup.
[17:20:50] [snowpack] lit-html/directives/class-map: Unscannable package import found.
Snowpack scans source files for package imports at startup, and on every change.
But, sometimes an import gets added during the build process, invisible to our file scanner.
We'll prepare this package for you now, but should add "lit-html/directives/class-map" to "knownEntrypoints"
in your config file so that this gets prepared with the rest of your imports during startup.
[17:20:50] [snowpack] lit-html/directives/style-map: Unscannable package import found.
Snowpack scans source files for package imports at startup, and on every change.
But, sometimes an import gets added during the build process, invisible to our file scanner.
We'll prepare this package for you now, but should add "lit-html/directives/style-map" to "knownEntrypoints"
in your config file so that this gets prepared with the rest of your imports during startup.
```

### Steps to Reproduce

1. clone my sample repo https://github.com/jaydanurwin/astro-lit-test
2. `npm i `
3. `npm start`
4. Notice that the `<mwc-button>` elements render as `[object Object]` on the page and the errors mentioned above are in the console.

### Link to Minimal Reproducible Example (Optional)

https://github.com/jaydanurwin/astro-lit-test

**Comment by jasikpark** [comment]
I seem to have gotten your code example working: https://codesandbox.io/s/recursing-sound-fz7ve?file=/src/pages/index.astro

Is this an intended change from https://github.com/snowpackjs/astro/releases/tag/astro%400.20.2 / https://github.com/snowpackjs/astro/pull/1178 @matthewp?

**Comment by jasikpark** [comment]
Without `hoisted` the script doesn't load on the dev server.

**Comment by jasikpark** [comment]
hmmm well it doesn't build.... 😰

```sh
sandbox@sse-sandbox-fz7ve:/sandbox$ yarn build --verbose
yarn run v1.22.11
$ npm run clean
npm WARN lifecycle The node binary used for scripts is /tmp/yarn--1631049127963-0.13345259266413123/node but npm is using /home/sandbox/.nvm/versions/node/v14.17.6/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.

> @example/framework-lit@0.0.1 clean /sandbox
> tsc --build --clean && rm -rf dist

$ tsc && astro build --verbose
[config] Set "buildOptions.site" to generate correct canonical URLs and sitemap
[22:13:18] [snowpack] Hint: run "snowpack init" to create a project config file. Using defaults...
[22:13:18] [snowpack] loaded plugin: /sandbox/node_modules/astro/snowpack-plugin-jsx.cjs
[22:13:18] [snowpack] loaded plugin: /sandbox/node_modules/astro/snowpack-plugin.cjs
[22:13:18] [snowpack] loaded plugin: /sandbox/node_modules/@snowpack/plugin-sass/plugin.js
[22:13:18] [snowpack] loaded plugin: /sandbox/node_modules/@snowpack/plugin-postcss/plugin.js
[22:13:18] [snowpack] Install directory ".meta" file is up-to-date. Welcome back!
[22:13:18] [snowpack] Scanning public/style/global.css for imports as CSS
[22:13:18] [snowpack] Scanning public/style/home.css for imports as CSS
[22:13:18] [snowpack] Scanning node_modules/astro/dist/frontend/500.astro for imports as Astro
[22:13:18] [snowpack] Scanning node_modules/astro/dist/frontend/__astro_config.js for imports as JS
[22:13:18] [snowpack] Scanning node_modules/astro/dist/frontend/hydrate/idle.js for imports asJS
[22:13:18] [snowpack] Scanning node_modules/astro/dist/frontend/hydrate/load.js for imports asJS
[22:13:18] [snowpack] Scanning node_modules/astro/dist/frontend/hydrate/media.js for imports as JS
[22:13:18] [snowpack] Scanning node_modules/astro/dist/frontend/hydrate/only.js for imports asJS
[22:13:18] [snowpack] Scanning node_modules/astro/dist/frontend/hydrate/visible.js for importsas JS
[22:13:18] [snowpack] Scanning src/components/Lorem.astro for imports as Astro
[22:13:18] [snowpack] Scanning src/components/Test.js for imports as JS
[22:13:18] [snowpack] Scanning src/pages/index.astro for imports as Astro
[22:13:18] [snowpack] Scanning src/components/my-counter-2/my-counter-2.js for imports as JS
[22:13:18] [snowpack] Scanning src/components/my-counter-2/my-counter-2.ts for imports as JS
[22:13:18] [snowpack] Scanning src/components/my-counter/my-counter.js for imports as JS
[22:13:18] [snowpack] Scanning src/components/my-counter/my-counter.ts for imports as JS
[22:13:19] [snowpack] Ready!
[22:13:19] [snowpack] Mounting directory: '/sandbox/public' as URL '/'
[22:13:19] [snowpack] Mounting directory: '/sandbox/node_modules/astro/dist/frontend' as URL '/_astro_frontend'
[22:13:19] [snowpack] Mounting directory: '/sandbox/src' as URL '/_astro/src'
[22:13:19] [snowpack] Using in-memory cache: [object Object]
[build] ! building pages...
[22:13:19] [snowpack-astro] transform() starting… [node_modules/astro/dist/frontend/__astro_config.js]
[22:13:19] [snowpack-astro] ✔ transform() success [node_modules/astro/dist/frontend/__astro_config.js]
[22:13:19] [@snowpack/postcss-transform] transform() starting… [node_modules/astro/dist/frontend/__astro_config.js]
[22:13:19] [@snowpack/postcss-transform] ✔ transform() success [node_modules/astro/dist/frontend/__astro_config.js]
[22:13:19] [snowpack] Scanning node_modules/astro/dist/frontend/__astro_config.js for imports as JS
[22:13:20] [snowpack-astro] load() starting… [src/pages/index.astro]
[22:13:20] [snowpack-astro] ✔ load() success [src/pages/index.astro]
[22:13:20] [snowpack-astro] transform() starting… [src/pages/index.js]
[22:13:20] [snowpack-astro] ✔ transform() success [src/pages/index.js]
[22:13:20] [@snowpack/postcss-transform] transform() starting… [src/pages/index.js]
[22:13:20] [@snowpack/postcss-transform] ✔ transform() success [src/pages/index.js]
[22:13:20] [snowpack] Scanning src/pages/index.astro for imports as JS
[22:13:20] [@snowpack/plugin-esbuild] load() starting… [src/components/my-counter/my-counter.ts]
[22:13:20] [snowpack-astro] load() starting… [src/components/Lorem.astro]
[22:13:20] [snowpack-astro] ✔ load() success [src/components/Lorem.astro]
[22:13:20] [snowpack-astro] transform() starting… [src/components/Lorem.js]
[22:13:20] [snowpack-astro] ✔ transform() success [src/components/Lorem.js]
[22:13:20] [@snowpack/postcss-transform] transform() starting… [src/components/Lorem.js]
[22:13:20] [@snowpack/postcss-transform] ✔ transform() success [src/components/Lorem.js]
[22:13:20] [snowpack] Scanning src/components/Lorem.astro for imports as JS
[22:13:20] [@snowpack/plugin-esbuild] ✔ load() success [src/components/my-counter/my-counter.ts]
[22:13:20] [snowpack-astro] transform() starting… [src/components/my-counter/my-counter.js]
[22:13:20] [snowpack-astro] ✔ transform() success [src/components/my-counter/my-counter.js]
[22:13:20] [@snowpack/postcss-transform] transform() starting… [src/components/my-counter/my-counter.js]
[22:13:20] [@snowpack/postcss-transform] ✔ transform() success [src/components/my-counter/my-counter.js]
[22:13:20] [snowpack] Scanning src/components/my-counter/my-counter.ts for imports as JS
[22:13:20] [snowpack] lit-element/lit-element.js: Unscannable package import found.
Snowpack scans source files for package imports at startup, and on every change.
But, sometimes an import gets added during the build process, invisible to our file scanner.
We'll prepare this package for you now, but should add "lit-element/lit-element.js" to "knownEntrypoints"
in your config file so that this gets prepared with the rest of your imports during startup.
[22:13:20] [snowpack] @lit/reactive-element/decorators/custom-element.js: Unscannable package import found.
Snowpack scans source files for package imports at startup, and on every change.
But, sometimes an import gets added during the build process, invisible to our file scanner.
We'll prepare this package for you now, but should add "@lit/reactive-element/decorators/custom-element.js" to "knownEntrypoints"
in your config file so that this gets prepared with the rest of your imports during startup.
[22:13:20] [snowpack] @lit/reactive-element/decorators/property.js: Unscannable package importfound.
Snowpack scans source files for package imports at startup, and on every change.
But, sometimes an import gets added during the build process, invisible to our file scanner.
We'll prepare this package for you now, but should add "@lit/reactive-element/decorators/property.js" to "knownEntrypoints"
in your config file so that this gets prepared with the rest of your imports during startup.
[22:13:20] [snowpack] @lit/reactive-element/decorators/state.js: Unscannable package import found.
Snowpack scans source files for package imports at startup, and on every change.
But, sometimes an import gets added during the build process, invisible to our file scanner.
We'll prepare this package for you now, but should add "@lit/reactive-element/decorators/state.js" to "knownEntrypoints"
in your config file so that this gets prepared with the rest of your imports during startup.
[22:13:20] [snowpack] @lit/reactive-element/decorators/event-options.js: Unscannable package import found.
Snowpack scans source files for package imports at startup, and on every change.
But, sometimes an import gets added during the build process, invisible to our file scanner.
We'll prepare this package for you now, but should add "@lit/reactive-element/decorators/event-options.js" to "knownEntrypoints"
in your config file so that this gets prepared with the rest of your imports during startup.
[22:13:20] [snowpack] @lit/reactive-element/decorators/query.js: Unscannable package import found.
Snowpack scans source files for package imports at startup, and on every change.
But, sometimes an import gets added during the build process, invisible to our file scanner.
We'll prepare this package for you now, but should add "@lit/reactive-element/decorators/query.js" to "knownEntrypoints"
in your config file so that this gets prepared with the rest of your imports during startup.
[22:13:20] [snowpack] @lit/reactive-element/decorators/query-all.js: Unscannable package import found.
Snowpack scans source files for package imports at startup, and on every change.
But, sometimes an import gets added during the build process, invisible to our file scanner.
We'll prepare this package for you now, but should add "@lit/reactive-element/decorators/query-all.js" to "knownEntrypoints"
in your config file so that this gets prepared with the rest of your imports during startup.
[22:13:20] [snowpack] @lit/reactive-element/decorators/query-async.js: Unscannable package import found.
Snowpack scans source files for package imports at startup, and on every change.
But, sometimes an import gets added during the build process, invisible to our file scanner.
We'll prepare this package for you now, but should add "@lit/reactive-element/decorators/query-async.js" to "knownEntrypoints"
in your config file so that this gets prepared with the rest of your imports during startup.
[22:13:20] [snowpack] @lit/reactive-element/decorators/query-assigned-nodes.js: Unscannable package import found.
Snowpack scans source files for package imports at startup, and on every change.
But, sometimes an import gets added during the build process, invisible to our file scanner.
We'll prepare this package for you now, but should add "@lit/reactive-element/decorators/query-assigned-nodes.js" to "knownEntrypoints"
in your config file so that this gets prepared with the rest of your imports during startup.
[22:13:20] [snowpack] + lit-element/lit-element.js@3.0.0-rc.2
[22:13:20] [snowpack] + @lit/reactive-element/decorators/custom-element.js@1.0.0-rc.2
[22:13:20] [snowpack] + @lit/reactive-element/decorators/property.js@1.0.0-rc.2
[22:13:20] [snowpack] + @lit/reactive-element/decorators/state.js@1.0.0-rc.2
[22:13:20] [snowpack] + @lit/reactive-element/decorators/event-options.js@1.0.0-rc.2
[22:13:20] [snowpack] + @lit/reactive-element/decorators/query.js@1.0.0-rc.2
[22:13:20] [snowpack] + @lit/reactive-element/decorators/query-async.js@1.0.0-rc.2
[22:13:20] [snowpack] + @lit/reactive-element/decorators/query-assigned-nodes.js@1.0.0-rc.2
[22:13:20] [snowpack] + @lit/reactive-element/decorators/query-all.js@1.0.0-rc.2
[build] ✔ pages built.
[build] built pages [607ms]
[22:13:20] [@snowpack/plugin-esbuild] load() starting… [src/components/my-counter-2/my-counter-2.ts]
[build] file:///sandbox/src/pages/index.astro: could not find "/_astro/src/pages/undefined"
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
sandbox@sse-sandbox-fz7ve:/sandbox$
```

**Comment by jaydanurwin** [comment]
Thanks for looking into this @jasikpark! Prior to 0.20 I wasn't having to add `hoisted` to my script tags. Was this added in that version?

**Comment by jasikpark** [comment]
It was a feature that was added to Astro, I don't think that it was meant to create a breaking change for you though!

**Comment by drwpow** [comment]
See also https://github.com/snowpackjs/astro/issues/1732

**Comment by FredKSchott** [comment]
Thanks for the report! We believe that this has been fixed in astro v0.21, going out today. Let us know if you can still reproduce the issue after updating to v0.21 and we'll be happy to take a look.
